api/src/main/java/graphql/nadel/hooks/ServiceExecutionHooks.java

```diff
@@ -46,4 +48,16 @@ default NewVariableValue visitArgumentValueInQuery(HooksVisitArgumentValueEnviro
     ) {
         return CompletableFuture.completedFuture(Optional.empty());
     }
+
+    /***
+     * Called to resolve the service that should be used to fetch data for a field that uses dynamic service resolution.
+     *
+     * @param services a list of all services registered on Nadel
+     * @param fieldName the name of the field
+     * @param arguments GraphQL arguments that were passed when querying the field
+     * @return the Service that should be used to fetch data for that field or an error that was raised when trying to resolve the service.
+     */
+    default ServiceOrError resolveServiceForField(List<Service> services, String fieldName, Map<String, Object> arguments) {
+        return null;
+    }
```

Comment:
There is no point in making this `default`

Nadel is not a stable Open Source lib - its a working 0.0.1-alpha that can break as need be.

By making it NOT default you will make sure we have to implement this once we integrate.

Also just fix up any tests

Reply:
I agree in general, but we have other methods here too which have a `default` impl, so it is not that we are consistent here.